lib/floe/workflow/choice_rule.rb

```diff
@@ -42,8 +44,18 @@ def true?(*)
 
       private
 
-      def variable_value(context, input)
-        Path.value(variable, context, input)
+      def validate_rule!(workflow)
+        if is_child?
+          invalid_field_error!("Next", @next, "not allowed in a child rule") if @next
+        else
+          missing_field_error!("Next") unless @next
+          invalid_field_error!("Next", @next, "is not found in \"States\"") if @next && !workflow_state?(@next, workflow)
+        end
+      end
+
+      # returns true if this is a child rule underneath an And/Or/Not
+      def is_child? # rubocop:disable Naming/PredicateName
+        %w[And Or Not].include?(name[-3])
```

Comment:
This looks a little fragile, maybe check array union `(name & %w[And Or Not]).empty?`

Reply:
ok, added a spec and this works:
```ruby
      def is_child? # rubocop:disable Naming/PredicateName
        !(%w[And Or Not] & name[0..-2]).empty?
      end
```

Kind of prefer `name[-3]`, though it takes some thinking. This other way requires less thinking. Which is always prefered

Reply:
Yeah thinking about it some more I like that name[-3] is more exact but I definitely did a triple take reading it. Maybe a comment about what the name array looks like so it is more obvious